### PR TITLE
Handle and warn properly for invalid opensearch xml.

### LINKED ISSUE #4162: Validator failed to provide a valid rejection on search engine without a valid <shortname>
### Describe the problem and steps to reproduce it:
Uploading a search engine XML file with an empty ShortName value will result to a bad behavior for the developer where he/she can't figure out what to fix to continue uploading to AMO.

### What happened?
Uploading the following XML :
[gists.zip](https://github.com/mozilla/addons-server/files/663875/gists.zip)

Will result with the following message 
"Your add-on failed validation with 1 error. Could not parse uploaded file."

The XML can be parsed as it's valid XML. However, the validator doesn't show any valid error for the user to fix his/her issue :
https://addons.mozilla.org/en-US/developers/upload/fcd799b5314c47308ca8b2b2b11dac01

### What did you expect to happen?
The validator should warn the developer that ShortName must not be empty.

### Anything else we should know?
I'm Spoji on IRC under #amo-editors if you got any questions regarding this bug. A developer called Jan\
needed help regarding this issue and I filled a bug for him/her.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #4162

amo-validator validates this but there are cases where we parse the XML
before amo-validator (e.g to get some meta-data) so let's handle that.

### CODE REVIEW

File: src/olympia/files/utils.py [hunk 1]
```diff
@@ -398,8 +398,12 @@ def extract_search(content):
...
     dom = minidom.parse(content)
 
-    def text(x):
-        return dom.getElementsByTagName(x)[0].childNodes[0].wholeText
+    def text(tag):
+        try:
+            return dom.getElementsByTagName(tag)[0].childNodes[0].wholeText
+        except (IndexError, AttributeError):
+            raise forms.ValidationError(
+                _('Could not parse uploaded file, missing <%s> element') % tag)
```
Comment (wagnerand) [inline src/olympia/files/utils.py]: `missing or empty`?

File: src/olympia/files/utils.py [hunk 2]
```diff
@@ -398,8 +398,12 @@ def extract_search(content):
...
 
     rv['name'] = text('ShortName')
     rv['description'] = text('Description')
```
Comment (wagnerand) [inline src/olympia/files/utils.py]: Are we sure the `description` is mandatory?

File: src/olympia/files/utils.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/files/utils.py]: An empty node is for us essentially missing so they're similar enough that I chose that wording. But yeah, maybe it's worth being even more explicit. I'll change that!

File: src/olympia/files/utils.py (hunk 2 above)
Comment (EnTeQuAk) [inline src/olympia/files/utils.py]: Always has been, or, we always failed if it didn't exist. Also, amo-validator validates that it exists.

**Pull request merged**